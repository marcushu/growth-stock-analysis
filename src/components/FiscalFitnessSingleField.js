import React from 'react'
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Form from 'react-bootstrap/Form';

import PropTypes from 'prop-types';

class FiscalFitnessSingleField extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      val: this.props.val
    }

    FiscalFitnessSingleField.protoTypes = {
      title: PropTypes.string,
      scoresId: PropTypes.string,
      val: PropTypes.number,
      updateField: PropTypes.func
    }
  }

  changeMe = (e) => {
    this.setState({ val: e.target.value }, 
      this.props.updateField({id: this.props.scoresId, val: e.target.value}));
  }

  render() {
    return (
      <Row>
        <Col></Col>
        <Col>
          <Form>
            <Form.Group>
              <Form.Label  className="font-weight-light">{this.props.title}</Form.Label>
              <Form.Control
                type="text"
                value={this.state.val}
                onChange={(e) => this.changeMe(e)}
                style={{ width: "150px", marginRight: "20px", border: "0", backgroundColor: "#E3E3E3"}}
              />
            </Form.Group>
          </Form>
        </Col>
        <Col></Col>
      </Row>
    )
  }
};

export default FiscalFitnessSingleField;