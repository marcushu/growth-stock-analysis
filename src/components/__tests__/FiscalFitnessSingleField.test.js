import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import FiscalFitnessSingleField from '../FiscalFitnessSingleField';

afterEach(cleanup);

describe("Fiscal fitness single field", () => {
 it('correctly renders label', () => {
    const { getByText } = render(
      <FiscalFitnessSingleField
        title="test_title"
        scoresId="test_scoresId"
        val={0}
        updateField={() => { }} />
    );

    expect(getByText('test_title')).toBeInTheDocument();
  });

 it('renders', () => {
    const { asFragment } = render(
      <FiscalFitnessSingleField
        title="test_title"
        scoresId="test_scoresId"
        val={0}
        updateField={() => { }} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

 it("calls update function when input changed", () => {
    const tstFunc = jest.fn();

    const { getByDisplayValue } = render(
      <FiscalFitnessSingleField
        title="test_title"
        scoresId="test_scoresId"
        val={0}
        updateField={tstFunc} />
    );

    fireEvent.change(getByDisplayValue("0"), { target: { value: 1 } });
    
    expect(tstFunc).toHaveBeenCalled();
  })
})



