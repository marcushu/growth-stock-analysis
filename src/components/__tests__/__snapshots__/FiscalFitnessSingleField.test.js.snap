// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fiscal fitness single field renders 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col"
    />
    <div
      class="col"
    >
      <form
        class=""
      >
        <div
          class="form-group"
        >
          <label
            class="font-weight-light form-label"
          >
            test_title
          </label>
          <input
            class="form-control"
            style="width: 150px; margin-right: 20px; border: 0px; background-color: rgb(227, 227, 227);"
            type="text"
            value="0"
          />
        </div>
      </form>
    </div>
    <div
      class="col"
    />
  </div>
</DocumentFragment>
`;
